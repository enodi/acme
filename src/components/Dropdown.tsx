import React from "react";

interface DropdownProps {
  filterFavourite: (event: React.ChangeEvent<HTMLSelectElement>) => void;
  favouriteConversation: string;
}

const Dropdown: React.FC<DropdownProps> = ({
  filterFavourite,
  favouriteConversation,
}) => (
  <select
    className="dropdown"
    value={favouriteConversation}
    onChange={filterFavourite}
  >
    <option value="conversations">All conversations</option>
    <option value="isFavorite">Favourites</option>
  </select>
);

export default Dropdown;
