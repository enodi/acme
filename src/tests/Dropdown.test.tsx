import React from "react";
import { shallow } from "enzyme";

import Dropdown from "../components/Dropdown";

let filterFavourite: (event: React.ChangeEvent<HTMLSelectElement>) => void,
  wrapper: any;
beforeEach(() => {
  filterFavourite = jest.fn();
  wrapper = shallow(
    <Dropdown
      filterFavourite={filterFavourite}
      favouriteConversation={"isFavorite"}
    />
  );
});

test("should render Dropdown component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should sort by isFavorite", () => {
  const value = "isFavorite";
  wrapper.find("select").simulate("change", { target: { value } });
  expect(filterFavourite).toHaveBeenCalled();
});
