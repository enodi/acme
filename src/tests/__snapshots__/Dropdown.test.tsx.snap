// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dropdown component correctly 1`] = `
<select
  className="dropdown"
  onChange={[MockFunction]}
  value="isFavorite"
>
  <option
    value="conversations"
  >
    All conversations
  </option>
  <option
    value="isFavorite"
  >
    Favourites
  </option>
</select>
`;
